### PR TITLE
unifying KafkaUtils.CreateDirectStream API

### PR DESCRIPTION
**Pull request by xiongrenyi**

### CODE REVIEW

File: csharp/Adapter/Microsoft.Spark.CSharp/Streaming/Kafka.cs [hunk 1]
```diff
@@ -170,5 +141,119 @@ public class KafkaUtils
...
+
+        /// <summary>
+        /// topics should contain only one topic if choose to repartitions to a configured numPartitions
+        /// TODO: move to scala and merge into DynamicPartitionKafkaRDD.getPartitions to remove above limitation
+        /// </summary>
+        /// <param name="ssc"></param>
+        /// <param name="topics"></param>
+        /// <param name="kafkaParams"></param>
+        /// <returns></returns>
+        private static int GetNumPartitionsFromConfig(StreamingContext ssc, List<string> topics, Dictionary<string, string> kafkaParams)
```
Comment (qintao1976) [inline csharp/Adapter/Microsoft.Spark.CSharp/Streaming/Kafka.cs]: >GetNumPartitionsFromConfig [](start = 27, length = 26)

Consider to change the name to something like "GetDynamicPartitionNum"?

**Comment by qintao1976** [comment]
LGTM

Please consider to do some validation in cluster before merge the PR.

Thanks

File: csharp/Adapter/Microsoft.Spark.CSharp/Streaming/Kafka.cs (hunk 1 above)
Comment (xiongrenyi) [inline csharp/Adapter/Microsoft.Spark.CSharp/Streaming/Kafka.cs]: this actually is 'static' version of repartition, maxMessagesPerTask based repartition is 'dynamic'

**Comment by xiongrenyi** [comment]
revert readFunc signature and use GetOffsetRange util instead, which is less impactful and more flexible

tested in cluster

**Comment by qintao1976** [comment]
LGTM

**Comment by xiongrenyi** [comment]
@qintao1976 Can you also verify on your side when you get a chance?

**Pull request merged**